test/perf/regression/target.go

```diff
@@ -0,0 +1,265 @@
+		return
+	}
+
+	t.logger.Infof("Deleted the namespace %s", t.namespace)
+}
+
+func (t *target) createNamespace() error {
+	t.logger.Infof("Creating namespace with prefix %s", t.namespacePrefix)
+	ns, err := t.typedClient.CoreV1().Namespaces().Create(&corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: t.namespacePrefix,
+		},
+	})
+
+	if err != nil {
+		return err
+	}
+
+	t.namespace = ns.Name
+	t.logger.Infof("Created namespace %s", t.namespace)
+	return nil
+}
+
+func (t *target) createResources() error {
+	t.logger.Infof("Creating resources in the namespace %s", t.namespace)
+	params := map[string]string{
+		"namespace":   t.namespace,
+		"etcdImage":   t.etcdImage,
+		"etcdbrImage": t.etcdbrImage,
+	}
+	for _, resourcePath := range t.resourcePaths {
+		t.logger.Infof("Creating resource from %s", resourcePath)
+
+		var (
+			u   *unstructured.Unstructured
+			gvr *schema.GroupVersionResource
+			ri  dynamic.ResourceInterface
+			err error
+		)
+		{
+			var renderedJSON []byte
+			{
+				templ, err := template.ParseFiles(resourcePath)
+				if err != nil {
+					return err
+				}
+
+				b := &bytes.Buffer{}
+				err = templ.Execute(b, params)
+				if err != nil {
+					return err
+				}
+
+				renderedJSON, err = yaml.YAMLToJSON(b.Bytes())
+				if err != nil {
+					return err
+				}
+			}
+
+			u = &unstructured.Unstructured{}
+			err = u.UnmarshalJSON(renderedJSON)
+			if err != nil {
+				return err
+			}
+		}
+		{
+			gvr, err = t.getGroupVersionResource(u)
+			if err != nil {
+				return err
+			}
+
+			nri := t.untypedClient.Resource(*gvr)
+			ri = nri.Namespace(u.GetNamespace())
+		}
+		u, err = ri.Create(u, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+		t.logger.Infof("Created resource %s/%s in the namespace %s", gvr.Resource, u.GetName(), u.GetNamespace())
+	}
+
+	return nil
+}
+
+func (t *target) getGroupVersionResource(u *unstructured.Unstructured) (*schema.GroupVersionResource, error) {
+	arl, err := t.discoveryClient.ServerResourcesForGroupVersion(u.GetAPIVersion())
+	if err != nil {
+		return nil, err
+	}
+
+	gvk := u.GroupVersionKind()
+	for i := range arl.APIResources {
+		ar := &arl.APIResources[i]
+		if ar.Kind == gvk.Kind {
+			return &schema.GroupVersionResource{
+				Group:    gvk.Group,
+				Version:  gvk.Version,
+				Resource: ar.Name,
+			}, nil
+		}
+	}
+
+	return nil, fmt.Errorf("Not GroupVersionResource found for GroupVersionKind: %s", gvk)
+}
+
+func (t *target) isPodRunning(podSelector string) (bool, error) {
+	t.logger.Infof("Starting watch on pod with selector %s in the namespace %s", podSelector, t.namespace)
+	podList, err := t.typedClient.CoreV1().Pods(t.namespace).List(metav1.ListOptions{
+		LabelSelector: podSelector,
+	})
+	if err != nil {
+		return false, err
+	}
+
+	for i := range podList.Items {
+		p := &podList.Items[i]
+		if p.Status.Phase != corev1.PodRunning {
+			return false, nil
+		}
+
+		nNotReady := len(p.Status.ContainerStatuses)
```

Comment:
```suggestion
		notReady := len(p.Status.ContainerStatuses)
```

Reply:
This is aesthetics. I would rather keep `nNotReady`, if that is alright.